docs: add doc for assertions & data contracts (#10029)
datahub-project/datahub: docs/managed-datahub/observe/assertions.md

# Assertions

:::note Contract Monitoring Support
Currently we support Snowflake, Databricks, Redshift, and BigQuery for out-of-the-box contract monitoring as part of Acryl Observe.
:::

> ## What is an Assertion

Reviewer: do we need this header?

> An assertion is **a data quality test that finds data that violate one or more specified rules.** These serve as the backbone of [Data Contracts](/docs/managed-datahub/observe/data-contract.md) – this is how we verify the contract is met.

Reviewer: ```suggestion
An assertion is **a data quality test that finds data that violates a specified rule.** Assertions serve as the building blocks of [Data Contracts](/docs/managed-datahub/observe/data-contract.md) – this is how we verify the contract is met.
```

## How to Create and Run Assertions

> Data quality tests (a.k.a. assertions) can be ingested from a 3rd party tool or run by Acryl.

Reviewer: yes, but they can also be created through the UI when using Acryl Cloud. also the 3rd party tools are more complex, and so that section should be second

### 3rd Party Runners

You can integrate 3rd party tools as follows:

- [DBT Test](/docs/generated/ingestion/sources/dbt.md#integrating-with-dbt-test)
- [Great Expectations](../../../metadata-ingestion/integration_docs/great-expectations.md)

If you opt for a 3rd party tool, it will be your responsibility to ensure the assertions are run based on the Data Contract spec stored in DataHub. With 3rd party runners, you can get the Assertion Change events by subscribing to our Kafka topic using the [DataHub Actions Framework](/docs/actions/README.md).

### Acryl Observe

For Acryl-provided assertion runners, we can deploy an agent in your environment to hit your sources and DataHub. Acryl Observe offers out-of-the-box evaluation of the following kinds of assertions:

- [Freshness](/docs/managed-datahub/observe/freshness-assertions.md) (SLAs)
- [Volume](/docs/managed-datahub/observe/volume-assertions.md)
- [Custom SQL](/docs/managed-datahub/observe/custom-sql-assertions.md)
- [Column](/docs/managed-datahub/observe/column-assertions.md)

> These can be defined through the DataHub API or the UI. With Acryl Observe, you can get the Assertion Change event by getting API events via [AWS EventBridge](/docs/managed-datahub/operator-guide/setting-up-events-api-on-aws-eventbridge.md) (the availability and simplicity of setup of each solution dependent on your current Acryl setup – chat with your Acryl representative to learn more).

Reviewer: this is a bit confusing - you can get assertion change events for both acryl-run and 3rd party ones. to do so, you use datahub-actions (oss) or eventbridge (acryl cloud)

probably want to move this section to "alerts"
